src/main/java/org/dita/dost/writer/ValidationFilter.java

```diff
@@ -190,7 +195,35 @@ private AttributesImpl validateReference(final String attrName, final Attributes
         final String href = atts.getValue(attrName);
         if (href != null) {
             try {
-                new URI(href);
+                final URI uri = new URI(href);
+                final URI abs = URLUtils.stripFragment(currentFile.resolve(uri)).normalize();
+                if (abs.getScheme() != null && abs.getScheme().equals("file")) {
+                    final File p = new File(abs);
+                    try {
+                        final File canFile = p.getCanonicalFile();
+                        final String absPath = p.getAbsolutePath();
+                        final String canPath = canFile.toString();
+                        if (!Objects.equals(absPath, canPath) && Objects.equals(absPath.toLowerCase(), canPath.toLowerCase())) {
+                            switch (processingMode) {
+                                case STRICT:
+                                    throw new RuntimeException(MessageUtils.getMessage("DOTJ082E", abs.toString()).setLocation(locator).toString());
+                                case SKIP:
+                                    logger.error(MessageUtils.getMessage("DOTJ082E", abs.toString()).setLocation(locator).toString() + ", using invalid value.");
+                                    break;
+                                case LAX:
+                                    final URI corrected = URLUtils.setFragment(currentFile.relativize(canFile.toURI()), uri.getFragment());
+                                    if (res == null) {
+                                        res = new AttributesImpl(atts);
+                                    }
+                                    res.setValue(res.getIndex(attrName), currentFile.toString());
+                                    logger.error(MessageUtils.getMessage("DOTJ082E", abs.toString()).setLocation(locator).toString() + ", using " + corrected + " value.");
```

Comment:
```suggestion
                                    logger.error(MessageUtils.getMessage("DOTJ082E", abs.toString()).setLocation(locator).toString() + ", using " + corrected + ".");
```

_(No need to add the word "value" here, showing the corrected path is enough.)_